### DISCUSSION TITLE
minItems and maxItems not present in 3.1 docs?

### DISCUSSION THREAD

**Discussion by mimkorn** [OAI/OpenAPI-Specification#2602] [Q&A] [answered]
Hi, 

I've noticed that when ⌘F thorugh 3.0.3 spec I can see `minItems` and `maxItems` mentioned in the documentation, but not in 3.1. I assume this has to do with 3.1 being aligned with JSON Schema and maxItems being part of JSON Schema. Thus I assume it is available still, only it's not mentioned in the OpenAPI documentation? If that's so, I'd like this confirmed. 

Also, I'd like to say that for people not deeply familiar with JSON Schema abilities, those who are writing OpenAPI mainly — we now have to resort to finding good documentation for JSON Schema and looking for what they define as available there, losing the convenience of looking through OpenAPI specification only. That makes it a bit harder to work with. Just mentioning in case it's something that hasn't been thought of, e.g. because for the insiders JSON Schema may be more obvious than for outsiders.

**Comment by MikeRalphson** [comment, marked as answer]
Yes, this is the reason, `minItems` and `maxItems` (and all JSON Schema keywords unaltered by OAS) are now defined solely by the JSON Schema specification. This allows us to reduce the length of the OAS, avoid repetition and inadvertent mistakes/differences. We appreciate having to read two specs is more work, but realistically it has always been the case that a working knowledge of the JSON Schema draft which the OAS references is necessary to fully utilitise OAS.

Both projects are working on more narrative documentation, so consumers who prefer not to read things written in specification language have an alternative, more instructional route available to them; for example: https://oai.github.io/Documentation/

**Reply by mimkorn** [reply]
Thanks a lot for confirming and clarifying this. May others benefit from your answer as well. Have a nice day!

**Comment by handrews** [comment]
Answered, closing.
